src/resolvers.js

```diff
@@ -299,8 +303,8 @@ export class NewInstance {
   * @param key The key to resolve/instantiate.
   * @return Returns an instance of NewInstance for the key.
   */
-  static of(key) {
-    return new NewInstance(key);
+  static of(key, ...dynamicDependencies: any[]) {
```

Comment:
Can you add another `@param` for the new parameter so that it will have a description in the documentation? I think that's the last thing I need to merge this.